lib/addMethod/index.js

```diff
@@ -1,8 +1,47 @@
+const _ = require('lodash');
+
+const addMethodREST = require('./addMethodREST');
+const addMethodSOAP = require('./addMethodSOAP');
+
+const INVALID_TYPE_ERROR_MESSAGE = '`type` must be strings \'REST\' or \'SOAP\'';
+function validateAndGetType ([ methodName, config ]) {
+	const { type } = config;
+
+	if (!_.isString(type)) {
+		throw new Error(`${methodName}: ${INVALID_TYPE_ERROR_MESSAGE}`);
+	}
+	const lowerCaseType = type.toLowerCase();
+	switch (lowerCaseType) {
+		case 'rest':
+		case 'soap':
+			return lowerCaseType;
+		default:
+			throw new Error(`${methodName}: ${INVALID_TYPE_ERROR_MESSAGE}`);
+	}
```

Comment:
it seems like this switch statement is just acting as an if statement to make sure its one of the 2 correct values, is there a reason this needs to be a switch? if its for future expansion, you could have the valid options in an array, and check if the type specified is in that array

Reply:
Yes, I did it like this for future expansion. 

Doing this using an array and something like `includes` is a valid way of doing it; but since the values are known/predefined (and it's not going to be too many), switch is fine for this. Also, doing something like `includes` means iteration has to occur before condition evaluation and then another condition so the default is an error, whereas with switch it is only condition evaluations.